django/db/models/sql/query.py

```diff
@@ -534,7 +534,14 @@ def get_aggregation(self, using, aggregate_exprs):
             # Queries with distinct_fields need ordering and when a limit is
             # applied we must take the slice from the ordered query. Otherwise
             # no need for ordering.
-            inner_query.clear_ordering(force=False)
+            if (
+                not inner_query.is_sliced
+                and not inner_query.distinct_fields
+                and inner_query.orderby_issubset_groupby
+            ):
+                inner_query.order_by = ()
+                inner_query.extra_order_by = ()
+                inner_query.default_ordering = False
```

Comment:
Should we still delegate to `clear_ordering` with `force=True` instead?

```suggestion
            if (
                not inner_query.is_sliced
                and not inner_query.distinct_fields
                and inner_query.orderby_issubset_groupby
            ):
                inner_query.clear_ordering(force=True)
```